src/main/java/pl/zaprogramujzycie/mma/controllers/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,86 @@
+package pl.zaprogramujzycie.mma.controllers;
+
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.crossstore.ChangeSetPersister;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import pl.zaprogramujzycie.mma.utils.exceptions.ErrorMessage;
+
+import java.util.Date;
+
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(ChangeSetPersister.NotFoundException.class)
+    public ResponseEntity<ErrorMessage> resourceNotFoundException(ChangeSetPersister.NotFoundException ex, WebRequest request) {
+        ErrorMessage message = new ErrorMessage(
+                HttpStatus.NOT_FOUND.value(),
+                new Date(),
+                ex.getMessage(),
+                request.getDescription(false));
+        return new ResponseEntity<>(message, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(HttpClientErrorException.Unauthorized.class)
+    public ResponseEntity<ErrorMessage> unauthorizedException (HttpClientErrorException.Unauthorized ex, WebRequest request) {
+        ErrorMessage message = new ErrorMessage(
+                HttpStatus.UNAUTHORIZED.value(),
+                new Date(),
+                ex.getMessage(),
+                request.getDescription(false));
+
+        return  new ResponseEntity<>(message, HttpStatus.UNAUTHORIZED);
+    }
+
+    @ExceptionHandler(HttpClientErrorException.Forbidden.class)
```

Comment:
Moze stworz customowe wyjatki zamiast tych? co to wgl za klasa, nie znam?

Reply:
Testowałam z grubsza, ale jeszcze zamierzam podłubać w testach, żeby te messages posprawdzać. Zwraca właściwe kody błędów. Jest to związane z hierarchią obsługi błędów: od szczegółu do ogółu. Ten handler łapie wszystkie, które nie zostały złapane po drodze, przez jakiś bardziej szczegółowy błąd.